src/models/release-asset.ts

```diff
@@ -0,0 +1,468 @@
+import type {
+	ODataOptionsWithoutCount,
+	OptionsToResponse,
+} from 'pinejs-client-core';
+import type {
+	InjectedDependenciesParam,
+	InjectedOptionsParam,
+	Release,
+	ReleaseAsset,
+} from '..';
+import { mergePineOptions } from '../util';
+import type { ReleaseRawVersionApplicationPair } from './release';
+import pLimit from 'p-limit';
+import { getType } from 'mime';
+
+const getMimeType = (filePath: string): string => {
+	return getType(filePath) ?? 'application/octet-stream';
+};
+
+const getFileBasename = (filePath: string): string => {
+	return filePath.split('/').pop() ?? filePath.split('\\').pop() ?? filePath;
```

Comment:
Optional: What do you think about v ?
```suggestion
	return filePath.match(/\/([^/]+)$/)?.[1] ?? filePath.match(/\\([^\\]+)$/)?.[1] ?? filePath;
```

Reply:
I think it is (unnecesarily) harder to read :) but I don't mind changing if you see great improvements